components/restapi/core/accounts.go

```diff
@@ -43,6 +43,16 @@ func congestionByAccountAddress(c echo.Context) (*api.CongestionResponse, error)
 		return nil, ierrors.Wrapf(httpserver.ErrInvalidParameter, "address %s is not an account address", c.Param(api.ParameterBech32Address))
 	}
 
+	maxCommittableAge := deps.RequestHandler.CommittedAPI().ProtocolParameters().MaxCommittableAge()
+	latestCommittedSlot := deps.RequestHandler.GetLatestCommitment().Slot()
+	if latestCommittedSlot >= maxCommittableAge && commitment.Slot()+maxCommittableAge < latestCommittedSlot && queryCommitmentID != iotago.EmptyCommitmentID {
+		return nil, ierrors.Wrapf(echo.ErrBadRequest, "invalid commitmentID, target slot index older than allowed (%d<%d)", commitment.Slot(), latestCommittedSlot-maxCommittableAge)
+	}
+
+	if commitment.Slot() > latestCommittedSlot && queryCommitmentID != iotago.EmptyCommitmentID {
+		return nil, ierrors.Wrapf(echo.ErrBadRequest, "invalid commitmentID, slot %d is not committed yet, latest committed slot: %d", commitment.Slot(), latestCommittedSlot)
+	}
+
```

Comment:
Move this check right after we parse queryCommitmentID, so we can return this error asap.

```go
if queryCommitmentID != iotago.EmptyCommitmentID {
    if commitment.Slot() > latestCommittedSlot {...}

    if latestCommittedSlot >= maxCommittableAge && commitment.Slot()+maxCommittableAge < latestCommittedSlot {...}

}
```